### PR TITLE
Fix misspell issues

### PR DESCRIPTION
**Pull request by mjtrangoni**
Hi,
I fixed here some trivial typos found by codespell and misspell.

### CODE REVIEW

**Review by marefr — CHANGES_REQUESTED** [review]
Thanks. Some minor comments

File: pkg/util/md5_test.go [hunk 1]
```diff
@@ -3,7 +3,7 @@ package util
 import "testing"
 
 func TestMd5Sum(t *testing.T) {
-	input := "dont hash passwords with md5"
+	input := "don't hash passwords with md5"
```
[the remark this answers is not in the document]
Comment (mjtrangoni) [reply pkg/util/md5_test.go]: fixed

File: devenv/dev-dashboards/panel_tests_polystat.json [hunk 2]
```diff
@@ -884,8 +884,8 @@
               "value": "celsius"
             },
             {
-              "text": "Farenheit (°F)",
-              "value": "farenheit"
+              "text": "Fahrenheit (°F)",
+              "value": "fahrenheit"
```
[the remark this answers is not in the document]
Comment (mjtrangoni) [reply devenv/dev-dashboards/panel_tests_polystat.json]: fixed

File: pkg/services/provisioning/datasources/testdata/broken-yaml/commented.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@
 #     org_id: 1
 
 # # list of datasources to insert/update depending
-# # whats available in the datbase
+# # what's available in the datbase
```
[the remark this answers is not in the document]
Comment (mjtrangoni) [reply pkg/services/provisioning/datasources/testdata/broken-yaml/commented.yaml]: done

**Comment by mjtrangoni** [comment]
@marefr done, should I rebase or is it ok so? Thanks!

**Review by marefr — APPROVED** [review]

**Comment by marefr** [comment]
Thank you for contributing to Grafana!

**Pull request merged**